pennylane/io/qasm_interpreter.py

```diff
@@ -0,0 +1,305 @@
+
+        Returns:
+            QuantumGate: The gate to execute.
+            list: The list of arguments to the QuantumGate.
+            list: The wires the gate applies to.
+        """
+        # setup arguments
+        args = []
+        for arg in node.arguments:
+            args.append(self.evaluate_argument(arg, context))
+
+        # retrieve gate method
+        gate = gates_dict[node.name.name.upper()]
+
+        # setup wires
+        wires = [
+            # parser will sometimes represent as a str and sometimes as an Identifier
+            (
+                node.qubits[q].name
+                if isinstance(node.qubits[q].name, str)
+                else node.qubits[q].name.name
+            )
+            for q in range(len(node.qubits))
+        ]
+
+        self._require_wires(wires, context)
+
+        if context["wire_map"] is not None:
+            wires = list(map(lambda wire: context["wire_map"][wire], wires))
+
+        return gate, args, wires
+
+    @staticmethod
+    def apply_modifier(mod: QuantumGate, previous: Operator, context: dict, wires: list):
+        """
+        Applies a modifier to the previous gate or modified gate.
+
+        Args:
+            mod (QASMNode): The modifier QASMNode.
+            previous (Operator): The previous (called) operator.
+            context (dict): The current context.
+            wires (list): The wires that the operator is applied to.
+
+        Raises:
+            NotImplementedError: If the modifier has a param of an as-yet unsupported type.
+        """
+        # the parser will raise when a modifier name is anything but the three modifiers (inv, pow, ctrl)
+        # in the OpenQASM 3.0 spec. i.e. if we change `pow(power) @` to `wop(power) @` it will raise:
+        # `no viable alternative at input 'wop(power)@'`, long before we get here.
+        assert mod.modifier.name in ("inv", "pow", "ctrl", "negctrl")
+        next = None
+
+        if mod.modifier.name == "inv":
+            next = ops.adjoint(previous)
+        elif mod.modifier.name == "pow":
+            if re.search("Literal", mod.argument.__class__.__name__) is not None:
+                power = mod.argument.value
+            elif (
+                "vars" in context
+                and hasattr(mod.argument, "name")
+                and mod.argument.name in context["vars"]
+            ):
+                power = context["vars"][mod.argument.name]["val"]
+            else:
+                raise NotImplementedError(
+                    f"Unable to handle {mod.argument.__class__.__name__} expression at this time"
+                )
+            next = ops.pow(previous, z=power)
+        elif mod.modifier.name == "ctrl":
+            next = ops.ctrl(previous, control=wires[-1])
+            wires = wires[:-1]
+        elif mod.modifier.name == "negctrl":
+            next = ops.ctrl(previous, control=wires[-1], control_values=[0])
+            wires = wires[:-1]
```

Comment:
```suggestion
            wires = wires[:-1]
      else:
           raise ValueError(f"Unknown modifier {mod}")
```
Might not be something we can really test, but a bit safer to have an `else` branch when we are setting up variables.

Reply:
I'll just have to add a `#pragma: no cover` here.